crates/mysten-metrics/src/thread_stall_monitor.rs

```diff
@@ -27,28 +66,41 @@ pub fn start_thread_stall_monitor() {
         return;
     }
 
+    let last_update: Arc<Mutex<Instant>> = Arc::new(Mutex::new(Instant::now()));
+
+    {
+        let last_update = last_update.clone();
+        std::thread::spawn(move || {
+            info!("Starting thread stall monitor watchdog thread");
+            let mut stall_duration = None;
+
+            loop {
+                std::thread::sleep(WATCHDOG_INTERVAL);
```

Comment:
optional: I'm not sure a separate WATCHDOG_INTERVAL is indeed. If it is, maybe add a comment on why it is computed that way.

Reply:
It is probably not necessary after thinking some more